### DISCUSSION TITLE
Anyone have with Hires Fix, problem since commit 63ca5c17e76438a24af866dac53efd585eb17ccc ?

### DISCUSSION THREAD

**Discussion by agentj01** [vladmandic/sdnext#1315] [Q&A]
commit 63ca5c17e76438a24af866dac53efd585eb17ccc
seems to break something on the denoising strength parameter during Hires Fix on "From Text" tab, have to increase it by 0.35 to reproduce old generation with the same parameters. I had a generation with a 0.15 denoising strength and redo the same generation with 0 denoising strength and they are just the same. That means it has like no effect until 0.3 or something. 

Reverting to commit 73015663530874af6c8539ed1180fbdf0c0a79b4 seems to fix it and should be able to reproduce old generations in the mean time if it is really an issue.

Anyone can confirm this ?

**Comment by vladmandic** [comment]
hires fix was previously picking up wrong sampler for second pass, now its corrected. just make sure you set sampler you actually want in settings -> samplers.

**Reply by agentj01** [reply]
Ho ok, May i ask what sampler it was picking up before ? (never touched this parameter it is on Euler A by default)
Also the adjustment of the denoising strength is also intended ?
I have to increase it by 0.15 to somewhat reproduce old generations. (problem it tends to change the image too much and add unwanted stuff)

Thank you for the great work.

**Comment by GalaxyTimeMachine** [comment]
There was a problem with the last Nvidia driver update. Myself, and several others, had to roll back to the previous version to resolve the whole system grinding to a crawl when it got to the Hires fix steps.
Explained here: https://github.com/vladmandic/automatic/discussions/1285

**Comment by monoinyo** [comment]
hires fix now seems to be having problems with most VAEs in some way, images are ending washed out and greyed. Anybody else experiencing this?

edit: i'm also back on 531 drivers btw

**Comment by ghost** [comment]
I had image quality problems too for quite a while especially tiled vae and tiled diffusion. High res fix was ugly most of the time too.

I now got the studio driver for my card and the problems are gone! I also git pulled but I think it was the driver

edit: still need a lot higher denoise in highres fix and something is off

**Reply by agentj01** [reply]
ho i wasnt imagining things. But now it is between sticking to an old commit preserving what we had or blindlessly git pulling because the fear of missing out some good stuff :/

**Reply by ghost** [reply]
Very strange, I did this: https://github.com/vladmandic/automatic/discussions/1264#discussioncomment-6099683

And highres fix is working as expected again

**Reply by agentj01** [reply]
Nice.
Sadly didnt work for me. I still have to increase the denoising strength from 0.15 to 0.4 or 0.5 to somewhat reproduce all generations. But ya high denoising also mean losing a lot of the base image. It seems setting a sampler for 2nd pass or not setting any just does nothing for me too. O_o

**Reply by zethfoxster** [reply]
i can confirm, if you have a second sampler set in the settings it will completely ignore what you have in the second pass sampler selection completely

**Reply by vladmandic** [reply]
You can confirm using which version? I just posed update for that this morning.

**Reply by zethfoxster** [reply]
as of like, 5 mins ago...18:05:21-369136 INFO     Version: 511a8cbb Fri Jul 21 15:04:27 2023 -0400
could be just the results of residual settings lingering. I still had a karrass sampler in there i set previously. I was testing the various changes you committed today, and found that it was still ignoring the second pass sampler for me, so i searched here for answers before opening a ticket on it...came across this post about #1264...I checked and i saw the sampler i had assigned a long time ago, set it to none.
hit generate and noticed immedately that it did used the correct dpm s karras as first pass and unipc as second. 
it still does nothing with the second pass prompt tho. or using a normal sd model as refiner, which is really what i was hoping it would do.... but ill take it :)

sidenote: maybe the one in the settings could go away now, since it can conflict with the one on the txt2img page and appears to be for setting the same thing?

**Reply by vladmandic** [reply]
Second prompt should work with refiner.
And you can combine SD1. 5 mode as base and SDXL refiner. But not other way around.

What's missing is denoising start and second prompt for base, that requires update to diffusers that's not released yet (I've tested it).

**Reply by zethfoxster** [reply]
I was actually meaning, use 1.5sd model with a 1.5sd inpaint model to refine(or even any other 1.5sd model). It allows me to set one as a refiner, even loads it into mem...just doesnt use it.

**Reply by vladmandic** [reply]
Oh, yeah, thats coming. I allow it to be loaded like that because it's exactly my intention to use it like that, just not enough time to do everything at once...

**Comment by SAC020** [comment]
Having a (somewhat?) similar issue. 

The live preview image during highres fix is quite crisp and colorful just up to 99%, but the finished image is washed out, lacks contrast and saturation. I don't understand what happens just as it hits 100%, or what's the difference between the live preview and the finished image. I would be more than happy to be able to reproduce the live preview process.

**Reply by BWDrum** [reply]
As of earlier today, this issue seems to have been resolved for me. Would be curious if your results have improved as well. It may be a coincidence, but I had switched from the Latent HiRes Upscaler to  R-ERGAN 4x+ and ran some gens and got much better results. Then I switched back to Latent Upscaler and now everything seems to work correctly. Probably caught an update today as well.
